appveyor.yml

```diff
@@ -1,30 +1,183 @@
-skip_tags: true
-clone_depth: 1
+# As config was originally based on an example by Olivier Grisel. Thanks!
+# https://github.com/ogrisel/python-appveyor-demo/blob/master/appveyor.yml
+clone_depth: 50
 
-os: Visual Studio 2015
```

Comment:
We still want to test MSVC as well, it needs to stay working. So instead of removing it all, I suggest to use a build matrix with a mix of the two build configs (without increasing the total number of builds).

Reply:
Which MSVC are you concerned about? This configuration should test 2008, 2010, and 2015. The "os" has nothing to do with that.

Reply:
Ralf - I'm guessing you mean a lapack-lite build, without OpenBLAS?

Reply:
No, I didn't mean lapack-lite, that's okay to only test on Linux I think. The ``install`` and ``build_script`` part here explicitly add ``MINGW`` to the path and in the whole file there are only references to Cygwin and MinGW. So if MSVC is still being used, it's very much non-obvious (plus it's unclear what the purpose of the MinGW stuff is). 

So either something is wrong (like MSVC being dropped) or this needs a lot of comments explaining what is going on.

Reply:
MinGW is used to compile the OpenBLAS library. MSVC is used to compile everything else.

Reply:
If you look at the CI, MinGW is invoked once during configuration (system_info) and once to compile OpenBLAS. Everything else is `cl.exe`.

Reply:
That's fine, but I don't want to have to inspect the actual CI runs to understand this config file. So I'll repeat my request for some comments (in particular inside the ``ps`` and ``build_script`` sections).

Reply:
IIUC, all the compilers are installed in the appveyor VM regardless, and Python generally finds MSVC based on registry entries rather than PATH.

Reply:
That's my understanding too.

Reply:
@pv @matthew-brett not sure if you're arguing against including more documentation or not. My comment is not particular to the line ``os: Visual Studio 2015``. There's a lot of opaque stuff in this PR that simply needs explaining.

Reply:
No, I'm just explaining why it works.

Reply:
Yes, more documentation would be good.

I hate to be annoying, but I have previously done the same work in the windows-wheel-builder repo - here's a PR : https://github.com/numpy/windows-wheel-builder/pull/3 in case it's helpful.